WordPress/gutenberg
# Color Controls: Merge color dropdown components

Related:
- https://github.com/WordPress/gutenberg/pull/34027


## What?
In https://github.com/WordPress/gutenberg/pull/34027, a new `ToolsPanelColorDropdown` component was introduced to facilitate color dropdowns being rendered within the `ToolsPanel` used by the color block supports slot. 

As a result of further [discussion](https://github.com/WordPress/gutenberg/pull/34027#discussion_r841646722), this PR looks to merge this `ToolsPanelColorDropdown` with the original `ColorGradientSettingsDropdown`. 

## Why?

Given the overlap, refactoring these to a single component, should reduce maintenance, highlight how color controls may need to be rendered differently in different contexts, and help ensure both cases are updated appropriately should the color control need changing.

## How?

- Updates the original `ColorGradientSettingsDropdown` component
- Adds a new prop to toggle between rendering as an `ItemGroup` (original component) or as a `ToolsPanelItem`
- Replaces use of `ToolsPanelColorDropdown` with the new merged `ColorGradientSettingsDropdown` component for the color block supports
- Relocates the color dropdown styles from the color block supports file to live with the original component styles in `packages/block-editor/src/components/colors-gradients/style.scss`

## Testing Instructions

1. Load the block editor, create a new post and add both a group and cover block
2. Select the group block and ensure that all the color controls in the sidebar still function and look correct
    (This tests the `ToolsPanelItem` version of the color dropdown component)
3. Make sure to test resetting the color controls via the `ToolsPanel` menu
4. Select the cover block ensuring that the color control in the Overlay sidebar panel is looking and working correctly.
    (This tests the original `ItemGroup` version of the color dropdown component)
5. Save the post and double-check the frontend applies color selections.


## Screenshots or screencast 

https://user-images.githubusercontent.com/60436221/161907959-82a9cd5e-dd5f-480d-b506-15c91d56d0e3.mp4
